storybook/story-components/doc-item.js

```diff
@@ -1,76 +1,78 @@
+import { size } from './platform-styles';
 
 const Divider = () => <View style={styles.verticalDivider} />;
 
 const createDescription = description => {
   const nodeList = React.Children.toArray(description);
-  let content;
-  if (nodeList.length === 1) {
-    content = <Text>{nodeList}</Text>;
-  } else {
-    content = insertBetween(() => <Divider key={Math.random()} />, nodeList);
-  }
-  return <Text style={styles.text}>{content}</Text>;
+  return typeof description === 'string' ? (
+    <Text>{description}</Text>
+  ) : (
+    insertBetween(() => <Divider key={Math.random()} />, nodeList)
+  );
 };
 
-const DocItem = ({ description, example = {}, name, typeInfo, label }) => (
+const DocItem = ({ description, example = {}, name, typeInfo, label, required, defaultValue }) => (
   <View style={styles.example}>
-    {name && (
-      <DocText style={styles.title}>
-        <PropText label={label} name={name} typeInfo={typeInfo} />
-      </DocText>
+    {!!name && (
+      <PropText label={label} name={name} typeInfo={typeInfo} required={required} defaultValue={defaultValue} />
     )}
-    {description && <View style={styles.description}>{createDescription(description)}</View>}
+    {!!description && <View style={styles.description}>{createDescription(description)}</View>}
     {(example.render || example.code) && (
       <View style={styles.renderBox}>
         <DocText style={styles.exampleText}>Example</DocText>
         {example.render && <View>{example.render()}</View>}
         {example.render && example.code && <View style={styles.verticalDivider} />}
-        {example.code && <Text style={styles.code}>{example.code}</Text>}
+        {example.code && <DocCode code={example.code} />}
       </View>
     )}
   </View>
 );
 
-const PropText = ({ label, name, typeInfo }) => (
-  <DocText>
-    {label && <Text style={[styles.label, label === 'web' && styles.webLabel]}>{label}</Text>}
-    <Text style={styles.propName}>{name}</Text>
-    {typeInfo && (
-      <Text>
-        {': '}
-        <Text style={styles.code}>{typeInfo}</Text>
-      </Text>
+const PropText = ({ label, name, typeInfo, required, defaultValue }) => (
+  <View>
+    <DocText style={styles.title}>
+      {label && <Text style={[styles.label, label === 'web' && styles.webLabel]}>{label}</Text>}
+      <Text style={styles.propName}>{name}</Text>
+      {typeInfo && (
+        <Text>
+          {': '}
+          <InlineCode code={typeInfo} />
+        </Text>
+      )}
+    </DocText>
+    {required !== undefined && (
+      <DocText>
+        Required: <InlineCode code={required ? 'true' : 'false'} />
+      </DocText>
     )}
-  </DocText>
+    {!!defaultValue && <DocText>Default value: {defaultValue}</DocText>}
+  </View>
 );
 
 const styles = StyleSheet.create({
-  code: {
-    fontFamily: 'monospace, monospace',
-    fontSize: '1rem',
-    lineHeight: '1.3125em'
-  },
   example: {
-    marginBottom: 'calc(1.5 * 1.3125rem)'
+    marginBottom: size.xlarge
```

Comment:
For styles, it would make sense if the repository sticks to a consistent style. If we use `emotion`, let's have everything use `emotion`.

Also, extracting styles to a separate file will make it easier for newcomers to the code to quickly scan through and see what's happening.

Reply:
I'm going to refine the styles further. Waiting to see what Peter is going to settle on first.

Reply:
Cool 👍  I figured this might be the case.